azurerm/internal/services/datafactory/data_factory_resource.go

```diff
@@ -70,12 +73,24 @@ func resourceDataFactory() *schema.Resource {
 							Required: true,
 							ValidateFunc: validation.StringInSlice([]string{
 								"SystemAssigned",
+								"UserAssigned",
 							}, false),
 						},
+
+						"user_identity_ids": {
+							Type:     schema.TypeSet,
```

Comment:
Also we might need this to be `TypeList` to be consistent with the rest of the provider. Also this might be why we are getting a non-empty plan during the acc test

Reply:
done!